src/zenml/client.py

```diff
@@ -37,19 +35,24 @@
 )
 from zenml.io import fileio
 from zenml.logger import get_logger
-from zenml.models import ComponentModel, FlavorModel, ProjectModel, StackModel
-from zenml.models.pipeline_models import PipelineModel
-from zenml.stack import Flavor
-from zenml.stack.stack_component import StackComponentConfig
 from zenml.utils import io_utils
 from zenml.utils.analytics_utils import AnalyticsEvent, track
 from zenml.utils.filesync_model import FileSyncModel
-from zenml.zen_stores.base_zen_store import DEFAULT_PROJECT_NAME, BaseZenStore
 
 if TYPE_CHECKING:
-    from zenml.models import HydratedStackModel, UserModel
-    from zenml.stack import Stack
-
+    from zenml.config.pipeline_configurations import PipelineSpec
+    from zenml.enums import StackComponentType
+    from zenml.models import (
+        ComponentModel,
+        FlavorModel,
+        HydratedStackModel,
+        PipelineModel,
+        ProjectModel,
+        StackModel,
+        UserModel,
+    )
```

Comment:
I think you may have overdone it here. Not sure I understand why removing the model imports makes the CLI or the client faster. The models are pretty thin in terms of libraries that they depend on. This is where I would have drawn the line.